pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -2019,7 +2034,20 @@ func TestUpdateNode(t *testing.T) {
 			manager.nodeStore.Add(c.oldNode)
 			c.ds.Spec.UpdateStrategy = *strategy
 			manager.dsStore.Add(c.ds)
-			syncAndValidateDaemonSets(t, manager, c.ds, podControl, 0, 0, 0)
+
+			expectedEvents := 0
+			if c.test == "Node Allocatable changed" {
```

Comment:
Please don't introduce such changes. This is very brittle. Instead of checking the test case name, add more parameters to the test cases and check those.

Reply:
Done.
Thanks for your review.